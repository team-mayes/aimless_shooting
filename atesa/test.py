# Test script to check my work on this code.

import sys

min_dist = 3

eligible = ['one.his_1','three.his_3','four.his_2']
historyfiles = ['one.his','two.his']

def handle_min_dist(indices, history_lines):
    if len(indices) >= 2:
        for i in range(len(indices) - 1):
            if indices[i + 1] - indices[i] < min_dist:
                try:
                    eligible.remove(history_lines[indices[i + 1]].split(' ')[0])
                    del indices[i + 1]
                    return True
                except ValueError:
                    sys.exit('Error: I broke something in t')
    return False


if min_dist > 1:  # only bother if min_dist is greater than one
    for filename in eligible:
        for history_filename in historyfiles:
            if history_filename in filename:
                history_lines = open(history_filename, 'r').readlines()
                indices = []  # initialize indices of matches
                index = 0  # initialize index to keep track
                for history_line in history_lines:
                    if history_line.split(' ')[0] in eligible:
                        indices.append(index)
                    index += 1
                cont = True
                while cont == True:  # to keep my indices from getting tangled up by deletions
                    cont = handle_min_dist(indices, history_lines)

print(eligible)
