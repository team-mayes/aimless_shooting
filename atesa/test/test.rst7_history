test.rst7_1 S
