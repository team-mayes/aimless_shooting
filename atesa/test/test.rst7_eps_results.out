0.7000000000000007 1.3000000000000003 1.016471772288
0.7000000000000007 1.3000000000000003 0.98965607166
0.7000000000000007 1.3000000000000003 1.024488771468
0.7000000000000007 1.3000000000000003 1.095836715776
0.7000000000000007 1.3000000000000003 1.10054893114
0.7000000000000007 1.3000000000000003 1.016471772288
0.7000000000000007 1.3000000000000003 0.98965607166
0.7000000000000007 1.3000000000000003 1.024488771468
0.7000000000000007 1.3000000000000003 1.095836715776
0.7000000000000007 1.3000000000000003 1.10054893114
0.7000000000000007 1.3000000000000003 1.120971909428
